src/init.gradle

```diff
@@ -88,6 +90,44 @@ rootProject {
         }
     }
 
+    def cycloneDxBOMTask = tasks.register("generateCycloneDxBom", org.cyclonedx.gradle.CycloneDxTask) {
+        mustRunAfter(dependencyTree)
+        doFirst {
+            configurations {
+                customDependencies
+            }
+
+            def dependencyTreeFile = rootProject.layout.buildDirectory.file("licenses/tree_all.txt")
+            def lines = dependencyTreeFile.get().asFile.readLines()
+
+            // Add dependencies from top level 'tree_all.txt' to the custom configuration
+            for (String line : lines) {
+                def dependency = line.trim()
+                if (dependency == "" || !dependency.contains(".")) { break }
+                dependencies {
+                    customDependencies dependency
```

Comment:
Very surprising this works. I think there should be a way to make this more dynamic, without having to rely on a doFirst. Basically the task needs to have as an input the output of the task which generates `tree_all.txt`, then the configuration dependencies need to use `addAllLater` with a provider.

I'm not against pushing this but I wouldn't be surprised it breaks anytime.